src/target/riscv/riscv-013.c

```diff
@@ -5418,16 +5418,22 @@ static int riscv013_step_or_resume_current_hart(struct target *target,
 		return ERROR_OK;
 	}
 
+	LOG_TARGET_ERROR(target, "unable to %s. dmstatus=0x%08x",
+		step ? "single-step" : "resume", dmstatus);
+
 	dm_write(target, DM_DMCONTROL, dmcontrol);
+	LOG_TARGET_ERROR(target, "  cancelling RESUMEREQ");
```

Comment:
(Just a comment - no code change requested)

After reading the code of this function, this part looks questionable. I am referring to the original code, not to your changes:

`dmcontrol.resumereq` bit is specified as W1 and that means only writing 1 has an effect. Writing 0 to this bit should not have any effect (pointless operation). This is true for both 0.13 and 1.0 versions of the debug spec.

This should not block this merge request, though. I just wanted to mention it. We may file an issue for it, if you like.

Reply:
Yeah... Thanks for the catch. 
This whole section where we try to restore the state is a mess. 

I'll double check and will probably remove this in a subsequent commit.

For now I just only to cleanup error messages.